morph_tool/marker_helper.py

```diff
@@ -0,0 +1,200 @@
+                        }
+                    )
+
+                else:
+                    logger.info(f"marker type {marker.type} not understood")
+            builder.plot(filename=str(filename))
+        else:
+            raise Exception("Only plotly available for now")
+
+    def to_dicts(self):
+        """Create a list of dicts from marker class."""
+        out_dict = {"morph_name": self._morph_name, "morph_path": self.morph_path, "markers": []}
+        for marker in self.markers:
+            _d = marker.to_dict()
+            del _d["morph_name"]
+            del _d["morph_path"]
+            out_dict["markers"].append(_d)
+        return out_dict
+
+    def save(self, filepath=".", filename=None):
+        """Save marker data.
+
+        Args:
+            filepath  (str): path to folder to save marker
+            filename (str): if provided,  path to marker file
+
+        TODO: append to the list if file exists.
+        """
+        if filename is None:
+            filename = (Path(filepath) / ("markers_" + self.morph_name)).with_suffix(".yaml")
+        with open(filename, "w") as f:
+            yaml.dump(self.to_dicts(), f)
+
+
+class Marker:
+    """Container of a single marker."""
+
+    def __init__(self, label, tpe, data, morph_name=None, morph_path=None, plot_style=None):
+        """Create a marker.
+
+        Args:
+            label (str): label of marker
+            tpe (str): marker types (points, line, plane, etc...)
+            data (str): markere data
+            morph_name (str): name of corresponding morphology
+            morph_path (str): path to corresponding morphology
+            plot_style (dict): custom dict for ploting arguments
+        """
+        self._label = label
+        self._tpe = tpe
+        self._data = data
+        self._morph_name = morph_name
+        self.morph_path = morph_path
+        self._set_plot_style(plot_style)
+
+        self._check_valid()
+
+    def _set_plot_style(self, plot_style):
+        """Set plotting style."""
+        self.plot_style = {}
+        if self._tpe == "points":
+            self.plot_style = {"color": "black", "width": 10}
+        if plot_style is not None:
+            self.plot_style.update(plot_style)
+
+    @property
+    def label(self):
+        """"""
+        return self._label
+
+    @property
+    def type(self):
+        """"""
+        return self._tpe
+
+    @property
+    def data(self):
+        """"""
+        return self._data
+
+    @property
+    def morph_name(self):
+        """"""
+        return self._morph_name
+
+    def _check_valid(self):
+        """Check if the given marker data is valid."""
+        if True:
+            return True
+        raise Exception("Given marker data is not valid.")
+
+    @property
+    def list_data(self):
+        """Returns data without numpy types.
+
+        WIP: so that we can yaml/json easily
+        """
+        return np.array(self._data, dtype=float).tolist()
```

Comment:
Why do you cast data to np.float before casting to list?

Reply:
not sure what is best here. Basically,  I want to ensure that there is no numpy dypes in there, othrwise, the yaml is ugly. 
.tolist() does this change, but only for 1d it seems.  There should be a cleaner way to do that, but I didn't look much into it. I don't want to use NumpyEncoder thingy, I prefer to ensure no numpy is hidden there.

Reply:
`tolist()` is supposed to convert any-dimension arrays, isn't it?

Reply:
yes, I changed this thing a little, by casting to ndarray upon creating the class